src/assisted_test_infra/download_logs/download_logs.py

```diff
@@ -212,6 +215,57 @@ def download_logs(
         log.removeHandler(log_handler)
 
 
+@JunitTestCase()
+def download_logs_kube_api(
```

Comment:
Maybe try to split this function into 2 or more smaller functions

Reply:
will move logic of must-gather for kube-api